The BattlegroundMgr is using the copy constructor for Battlegrounds in CreateNewBattleground. I think that's why I'm unable to have a vector of unique_ptr's?
```
error C2280: 'std::unique_ptr<WorldPackets::Battleground::BattlegroundPlayerPosition,std::default_delete<WorldPackets::Battleground::BattlegroundPlayerPosition>>::unique_ptr(const std::unique_ptr<WorldPackets::Battleground::BattlegroundPlayerPosition,std::default_delete<WorldPackets::Battleground::BattlegroundPlayerPosition>> &)': attempting to reference a deleted function (compiling source file H:\Trinity\source\src\server\game\Battlegrounds\BattlegroundMgr.cpp)
````

--- a/src/server/game/Battlegrounds/Battleground.cpp
+++ b/src/server/game/Battlegrounds/Battleground.cpp
@@ -138,8 +138,6 @@ Battleground::~Battleground()
 
     for (BattlegroundScoreMap::const_iterator itr = PlayerScores.begin(); itr != PlayerScores.end(); ++itr)
         delete itr->second;
-
-    _playerPositions.clear();
 }
 
 void Battleground::Update(uint32 diff)
@@ -251,13 +249,13 @@ void Battleground::_ProcessPlayerPositionBroadcast(uint32 diff)
 
         WorldPackets::Battleground::BattlegroundPlayerPositions playerPositions;
 
-        for (std::shared_ptr<WorldPackets::Battleground::BattlegroundPlayerPosition> position : _playerPositions)
+        for (WorldPackets::Battleground::BattlegroundPlayerPosition& playerPosition : _playerPositions)
         {
             // Update position data if we found player.
-            if (Player* player = ObjectAccessor::GetPlayer(GetBgMap(), position->Guid))
-                position->Pos = player->GetPosition();
+            if (Player* player = ObjectAccessor::GetPlayer(GetBgMap(), playerPosition.Guid))
+                playerPosition.Pos = player->GetPosition();
 
-            playerPositions.FlagCarriers.push_back(*position);
+            playerPositions.FlagCarriers.push_back(playerPosition);
         }
 
         SendPacketToAll(playerPositions.Write());
@@ -1680,16 +1678,16 @@ void Battleground::PSendMessageToAll(uint32 entry, ChatMsg msgType, Player const
     va_end(ap);
 }
 
-void Battleground::AddPlayerPosition(std::shared_ptr<WorldPackets::Battleground::BattlegroundPlayerPosition> position)
+void Battleground::AddPlayerPosition(WorldPackets::Battleground::BattlegroundPlayerPosition const position)
 {
-    _playerPositions.push_back(std::move(position));
+    _playerPositions.push_back(position);
 }
 
 void Battleground::RemovePlayerPosition(ObjectGuid guid)
 {
-    auto const itr = std::remove_if(_playerPositions.begin(), _playerPositions.end(), [guid](std::shared_ptr<WorldPackets::Battleground::BattlegroundPlayerPosition> const playerPosition)
+    auto itr = std::remove_if(_playerPositions.begin(), _playerPositions.end(), [guid](WorldPackets::Battleground::BattlegroundPlayerPosition const info)
     {
-        return playerPosition->Guid == guid;
+        return info.Guid == guid;
     });
 
     _playerPositions.erase(itr, _playerPositions.end());

--- a/src/server/game/Battlegrounds/Battleground.cpp
+++ b/src/server/game/Battlegrounds/Battleground.cpp
@@ -140,6 +140,8 @@ Battleground::~Battleground()
 
     for (BattlegroundScoreMap::const_iterator itr = PlayerScores.begin(); itr != PlayerScores.end(); ++itr)
         delete itr->second;
+
+    _playerPositions.clear();
 }
 
 void Battleground::Update(uint32 diff)
@@ -1003,6 +1005,8 @@ void Battleground::Reset()
     PlayerScores.clear();
 
     ResetBGSubclass();
+
+    _playerPositions.clear();
 }
 
 void Battleground::StartBattleground()

--- a/src/server/game/Battlegrounds/Battleground.cpp
+++ b/src/server/game/Battlegrounds/Battleground.cpp
@@ -112,6 +112,8 @@ Battleground::Battleground(BattlegroundTemplate const* battlegroundTemplate) : _
     StartMessageIds[BG_STARTING_EVENT_FOURTH] = BG_TEXT_BATTLE_HAS_BEGUN;
 }
 
+Battleground::Battleground(Battleground const&) = default;
+
 Battleground::~Battleground()
 {
     // remove objects and creatures

--- a/src/server/game/Battlegrounds/Battleground.h
+++ b/src/server/game/Battlegrounds/Battleground.h
@@ -260,6 +260,7 @@ class TC_GAME_API Battleground
 {
     public:
         Battleground(BattlegroundTemplate const* battlegroundTemplate);
+        Battleground(Battleground const&);
         virtual ~Battleground();
 
         void Update(uint32 diff);

--- a/src/server/game/Battlegrounds/Battleground.h
+++ b/src/server/game/Battlegrounds/Battleground.h
@@ -23,7 +23,6 @@
 #include "Position.h"
 #include "SharedDefines.h"
 #include <map>
-#include <memory>
 #include <vector>
 
 class BattlegroundMap;
@@ -504,7 +503,7 @@ class TC_GAME_API Battleground
         // because BattleGrounds with different types and same level range has different m_BracketId
         uint8 GetUniqueBracketId() const;
 
-        void AddPlayerPosition(std::shared_ptr<WorldPackets::Battleground::BattlegroundPlayerPosition> position);
+        void AddPlayerPosition(WorldPackets::Battleground::BattlegroundPlayerPosition const position);
         void RemovePlayerPosition(ObjectGuid guid);
 
     protected:
@@ -629,6 +628,6 @@ class TC_GAME_API Battleground
         BattlegroundTemplate const* _battlegroundTemplate;
         PVPDifficultyEntry const* _pvpDifficultyEntry;
 
-        std::vector<std::shared_ptr<WorldPackets::Battleground::BattlegroundPlayerPosition>> _playerPositions;
+        std::vector<WorldPackets::Battleground::BattlegroundPlayerPosition> _playerPositions;
 };
 #endif

--- a/src/server/game/Spells/Auras/SpellAuraEffects.cpp
+++ b/src/server/game/Spells/Auras/SpellAuraEffects.cpp
@@ -6383,19 +6383,19 @@ void AuraEffect::HandleBattlegroundPlayerPosition(AuraApplication const* aurApp,
 
     if (apply)
     {
-        std::shared_ptr<WorldPackets::Battleground::BattlegroundPlayerPosition> playerPosition = std::make_shared<WorldPackets::Battleground::BattlegroundPlayerPosition>();
-        playerPosition->Guid = target->GetGUID();
-        playerPosition->ArenaSlot = static_cast<uint8>(GetMiscValue());
-        playerPosition->Pos = target->GetPosition();
+        WorldPackets::Battleground::BattlegroundPlayerPosition playerPosition;
+        playerPosition.Guid = target->GetGUID();
+        playerPosition.ArenaSlot = static_cast<uint8>(GetMiscValue());
+        playerPosition.Pos = target->GetPosition();
 
         if (GetAuraType() == SPELL_AURA_BATTLEGROUND_PLAYER_POSITION_FACTIONAL)
-            playerPosition->IconID = target->GetTeam() == ALLIANCE ? PLAYER_POSITION_ICON_HORDE_FLAG : PLAYER_POSITION_ICON_ALLIANCE_FLAG;
+            playerPosition.IconID = target->GetTeam() == ALLIANCE ? PLAYER_POSITION_ICON_HORDE_FLAG : PLAYER_POSITION_ICON_ALLIANCE_FLAG;
         else if (GetAuraType() == SPELL_AURA_BATTLEGROUND_PLAYER_POSITION)
-            playerPosition->IconID = target->GetTeam() == ALLIANCE ? PLAYER_POSITION_ICON_ALLIANCE_FLAG : PLAYER_POSITION_ICON_HORDE_FLAG;
+            playerPosition.IconID = target->GetTeam() == ALLIANCE ? PLAYER_POSITION_ICON_ALLIANCE_FLAG : PLAYER_POSITION_ICON_HORDE_FLAG;
         else
             TC_LOG_WARN("spell.auras", "Unknown aura effect %u handled by HandleBattlegroundPlayerPosition.", GetAuraType());
 
-        bg->AddPlayerPosition(std::move(playerPosition));
+        bg->AddPlayerPosition(playerPosition);
     }
     else
         bg->RemovePlayerPosition(target->GetGUID());

